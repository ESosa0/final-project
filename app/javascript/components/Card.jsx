import React from 'react'
import Line from '../../assets/images/line.svg'
import classNames from 'classnames'


class Card extends React.Component {
  render(){
    return (
      <li className="card">
        <img className="line" src={Line}/>
        <div className="box">
          <div className="question-step">
            {this.props.step}
          </div>
          <h2 className="question-text">
            {this.props.text}
          </h2>
          <p>
            {this.props.blurb}
          </p>
          <div className="choices">
            {this.props.children}
          </div>
          <button 
            className={classNames("next-button", {
              "next-button--disabled": this.props.preventsNextClick
            })}  
            type="button" 
            onClick={this.props.onNextClick} 
            disabled={this.props.preventsNextClick}
            >
            {this.props.buttonText}
          </button>
        </div>
      </li>
    )
  }
}

export default Card 


