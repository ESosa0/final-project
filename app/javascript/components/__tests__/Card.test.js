import Card from '../Card'
import {shallow} from 'enzyme'
import toJSON from 'enzyme-to-json'
import React from 'react'

describe("Card Component", () => {
  it("renders a default card", () => {
    const component = shallow(<Card/>)
    expect(toJSON(component)).toMatchSnapshot()
  })
})
