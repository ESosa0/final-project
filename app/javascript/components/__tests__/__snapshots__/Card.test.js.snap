// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component renders a default card 1`] = `
<li
  className="card"
>
  <img
    className="line"
    src="test-file-stub"
  />
  <div
    className="box"
  >
    <div
      className="question-step"
    />
    <h2
      className="question-text"
    />
    <p />
    <div
      className="choices"
    />
    <button
      className="next-button"
      type="button"
    />
  </div>
</li>
`;
